WordPress/Classes/ViewRelated/Blog/QuickStartPromptViewController.swift

```diff
@@ -101,25 +101,34 @@ final class QuickStartPromptViewController: UIViewController {
     // MARK: - IBAction
 
     @IBAction private func showMeAroundButtonTapped(_ sender: Any) {
-        dismissPrompt { [weak self] in
-
-            DispatchQueue.main.asyncAfter(deadline: .now() + Constants.quickStartDelay) {
-                guard let self = self else {
-                    return
+        if let onDismissEpilogue = onDismissEpilogue {
+            onDismissEpilogue()
+
+            // Show the My Site screen for the specified blog
+            WordPressAppDelegate.shared?.windowManager.dismissFullscreenSignIn(blogToShow: blog, completion: {
```

Comment:
This code is perhaps making assumptions about how this screen is being displayed and that full screen sign-in is being displayed. I wonder if we would be better to remove this login specific code from this VC and make it part of the onDismiss closure itself.

So in `WordPressAuthenticationManager.swift` we'd have something like:

```
let onDismiss = { showQuickStart in
    // Show the My Site screen for the specified blog
    WordPressAppDelegate.shared?.windowManager.dismissFullscreenSignIn(blogToShow: blog, completion: {
        guard showQuickStart else{
            return
        }

        // After a short delay, trigger the Quick Start tour
        DispatchQueue.main.asyncAfter(deadline: .now() + Constants.quickStartDelay) {
            QuickStartTourGuide.shared.setup(for: self.blog)
        }
    })
}

...

quickstartPrompt.onDismissEpilogue = onDismiss
```

That way, the quick start prompt doesn't need to know about the context in which it's displayed, or the complexities of how it should dismiss itself. What do you think?

Reply:
Yes, 100%. That's a very good point, will fix this.